tests/test_server.py

```diff
@@ -51,6 +52,7 @@
     removeTestTempDir,
 )
 from lsst.resources.http import HttpResourcePath
+from lsst.resources.s3utils import clean_test_environment_for_s3, getS3Client
```

Comment:
S3 dependencies are not installed by default in the pyproject.toml definition so if server testing is now required to have boto3 available this is going to have to have import protection so the tests can be skipped.